types/next/document.d.ts

```diff
@@ -1,6 +1,19 @@
 import * as React from "react";
 import * as http from "http";
 
+export interface RenderPageResponse {
+    buildManifest: { [key: string]: any };
+    chunks: {
+        names: string[];
+        filenames: string[];
+    };
+    html?: string;
+    head: Array<React.ReactElement<any>>;
+    errorHtml: string;
+}
+
+export type Enhancer<P = any> = (page: React.ComponentType<P>) => React.ComponentType<P>;
```

Comment:
Just curious, but is the method guaranteed `ComponentType` with propTypes `P`?  Can my enhancer change the PropTypes?  If not, should we (a) update it to be more explicit here or (b) make either or both of these only take `any`,  and only be restrictive where we need to be?

Reply:
For that matter, what do you think @disintegrator?